### PR TITLE
Add metrics for the rustdoc rendering times

### PR DESCRIPTION
**Pull request by pietroalbini**
I have a feeling the next refactoring I'm going to make (using a shared `Storage` instance for the whole application) is going to improve performance of rustdoc pages, so this adds some code to expose a metric that breaks down that endpoint's rendering time.

That endpoint is the one I think we should care about the most, so this is also going to be helpful to drive other performance improvements in the future.

### CODE REVIEW

File: src/web/metrics.rs [hunk 1]
```diff
@@ -197,6 +204,47 @@ impl iron::Handler for RequestRecorder {
...
+    pub(crate) fn step(&mut self, step: &'static str) {
+        self.record_current();
+        self.current = Some(RenderingTime {
+            start: Instant::now(),
+            step,
+        });
+    }
+
+    fn record_current(&mut self) {
+        if let Some(current) = self.current.take() {
```
Comment (jyn514) [inline src/web/metrics.rs:234]: This doesn't count the time to record the metric, is that intentional? If not you could make it (mostly) atomic like this:
```suggestion
        if let Some(current) = self.current.replace(Instant::now()) {
```
and then removing `current = ...` in `step()`.

Comment (pietroalbini) [reply src/web/metrics.rs:234]: Is that really needed? The thing I want to get out of this metrics is how much it takes to do database queries, fetch files from S3 and parse the HTML, and the time it takes to record metrics won't influence that.

**Comment by pietroalbini** [comment]
> Can you post some screenshots of what this looks like in grafana?

I don't have a (working) Grafana setup locally.

File: src/web/metrics.rs (hunk 1 above)
Comment (jyn514) [reply src/web/metrics.rs:234]: Ok, if it's not valuable we don't need to include it.

**Comment by jyn514** [comment]
Going to make a separate PR for a `prometheus` instance in docker-compose. This PR looks fine.

**Pull request merged**